src/cpp/shared/combat/combat_ai.cpp

```diff
@@ -0,0 +1,177 @@
+          incapacitated = true;
+      }
+
+      int effectValue = 0;
+      switch(spell) {
+        case SPELL_SLOW:
+        case SPELL_MASS_SLOW:
+          if(!incapacitated) {
+            if(!underMyControl) {
+              if(!potTarg->effectStrengths[EFFECT_SLOW]) {
+                effectValue = this->RawEffectSpellInfluence(potTarg, EFFECT_SLOW) * durBenefit;
+                if(potTarg->effectStrengths[EFFECT_HASTE]) {
+                  int durModifierIdx = targSpentTurn + potTarg->effectStrengths[EFFECT_HASTE];
+                  if(durModifierIdx >= 10)
+                    durModifierIdx = 10;
+
+                  effectValue -= this->RawEffectSpellInfluence(potTarg, EFFECT_HASTE) * heuristicModifierForDuration[durModifierIdx];
+                }
+              }
+            }
+          }
+          break;
+        case SPELL_CURSE:
+        case SPELL_MASS_CURSE:
+          if(!incapacitated) {
+            if(!underMyControl) {
+              if(!potTarg->effectStrengths[EFFECT_CURSE]) {
+                effectValue = -this->RawEffectSpellInfluence(potTarg, EFFECT_CURSE) * durBenefit;
+                if(potTarg->effectStrengths[EFFECT_BLESS]) {
+                  int durModifierIdx = targSpentTurn + potTarg->effectStrengths[EFFECT_BLESS];
+                  if(durModifierIdx >= 10)
+                    durModifierIdx = 10;
+                  effectValue -= this->RawEffectSpellInfluence(potTarg, EFFECT_BLESS) * heuristicModifierForDuration[durModifierIdx];
+                }
+              }
+            }
+          }
+          break;
+        default:
+          *value = 0;
+          break;
+      }
+
+      int finalEffectValue = 0;
+      switch(aiSpellType) {
+        case AI_MASS_BUFF_SPELL:
+        case AI_MASS_DEBUFF_SPELL:
+          finalEffectValue += effectValue;
+          break;
+        case AI_BATTLEFIELD_AFFECTING_SPELL:
+        case AI_BALL_SPALL:
+        case AI_TARGET_BUFF_SPELL:
+        case AI_TARGET_DAMAGE_OR_DEBUFF_SPELL:
+        case AI_RESURRECT_SPELL:
+        case AI_DISPEL_SPELL:
+          finalEffectValue = effectValue;
+          break;
+        default:
+          break;
+      }
+
+      if(aiSpellType == AI_MASS_BUFF_SPELL || aiSpellType == AI_MASS_DEBUFF_SPELL || *value < finalEffectValue) {
+        *value = finalEffectValue;
+        *target = hex;
+      }
+
+      switch(aiSpellType) {
+        case AI_BATTLEFIELD_AFFECTING_SPELL:
+          targetHex = 1;
+          break;
+        case AI_MASS_BUFF_SPELL:
+        case AI_MASS_DEBUFF_SPELL:
+        case AI_TARGET_BUFF_SPELL:
+        case AI_TARGET_DAMAGE_OR_DEBUFF_SPELL:
+        case AI_DISPEL_SPELL:
+          targetHex = this->FirstArmy(hex + 1, targetSide, &hex);
+          break;
+        case AI_RESURRECT_SPELL:
+          targetHex = this->FirstResurrectable(hex + 1, &hex, spell);
+          break;
+        case AI_BALL_SPALL:
+          this->NextPos(&hex);
+          if(hex > 43)
```

Comment:
Do you know what this is for?

Reply:
for area spells fireball, fireblast, meteorshower and coldring

Reply:
No, I mean: what's special about hex 43?